app/javascript/packages/document-capture/components/documents-step.tsx

```diff
@@ -109,6 +116,9 @@ function DocumentsStep({
   const { flowPath } = useContext(UploadContext);
   const { exitQuestionSectionEnabled } = useContext(FeatureFlagContext);
   const { isSelfieCaptureEnabled } = useContext(SelfieCaptureContext);
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  logDeviceResolution(trackEvent);
```

Comment:
should this be in a `useEffect` block or something to make sure it doesn't re-fire multiple times?

Reply:
also it seems a little weird to pass in `trackEvent`, could this function be rewritten to be able to reference the `AnalyticsContext` itself?

Reply:
> also it seems a little weird to pass in trackEvent
It is definitely odd. The reason I'm doing it like this is basically because when I try and convert it to use `trackEvent` in the new file it turns into a lot of changes and also makes the new file much more complex with `imports` at the top. As it is, this is a pretty simple new file, but moving `trackEvent` in there gets complicated fast.

Reply:
> should this be in a useEffect block 

Yep! I hadn't gotten there, but it's in one now.

Reply:
If we wrote this as a custom hook, we could avoid passing in the `trackEvent`, since the hook could reference the context itself? Also has the upside of fitting a bit cleaner into React conventions, vs. new services file.

```ts
function useCameraInfoLogger() {
  const { trackEvent } = useContext(AnalyticsContext);
  // ...
}
```

Reply:
👍 I'll convert it to a custom hook.

I did start converting to a custom hook, but wasn't sure the time investment was worth it. Given this feedback, I'll revisit and do that conversion.